### DISCUSSION TITLE
Villain payload

### DISCUSSION THREAD

**Discussion by elchinsir** [t3l3machus/Villain#179] [General]
Hi everyone, I am trying to use Villain tool, I sued to have no problem but now when i do this "generate os=windows lhost=eth0 obfuscate" it says argument PAYLOAD not supplied and when I do "generate payload=windows/reverse_tcp/powershell_reverse_tcp lhost=eth0 " this it says "Payload template not found.". Any help?

**Comment by t3l3machus** [comment]
@elchinsir looks like you haven't used Villain for a while. The syntax of certain commands has changed a lot in recent releases. Please, read the help messages again. When generating reverse shell commands, you can use tab autocomplete to see your options, most reverse tcp templates have different names. For example, the syntax for the command you are trying to generate is:
```
generate payload=windows/reverse_tcp/powershell lhost=eth0
```

List all available templates with tab autocomplete:
![image](https://github.com/user-attachments/assets/fc7b1175-428f-46ba-90e2-49c64a0e3922)

**Comment by marques-thanatos** [comment]
Hi t3l3machus. The installation of Villain does not work anymore. I tried both on Kali and on Ubuntu. But everytime i try to execute sudo python3 Villain.py, i'm getting this: "File "/home/machine1/Villain/Villain.py", line 11, in <module>
    from Core.common import *
  File "/home/machine1/Villain/Core/common.py", line 15, in <module>
    from Cryptodome.Cipher import AES
ModuleNotFoundError: No module named 'Cryptodome' " i searched all over the internet to fix this error but none worked it out. Can you give me a hand? It's totally strange because 2 years ago i used Villain on my Ubuntu and it worked perfectly.

**Comment by t3l3machus** [comment]
Probably the required packages are not available in the root user’s Python environment. You need to install requirements.txt with sudo pip3

**Comment by marques-thanatos** [comment]
I really did install using sudo pip3 install -r requirements.txt , but the problem remains, both on Ubuntu and Kali
